redis/src/cluster_async/mod.rs

```diff
@@ -146,16 +146,28 @@ impl<C> CmdArg<C> {
         }
 
         match self {
-            Self::Cmd { ref cmd, .. } => route_for_command(cmd),
+            Self::Cmd { ref cmd, .. } => {
+                let value = parse_redis_value(cmd).expect("Command wasn't a valid redis value");
```

Comment:
These expects are a bit scary. Is it possible to handle them earlier in the request / without panicking?

Reply:
Mapped them to None for now. I believe this should be an error, but since I'm refactoring this area in https://github.com/redis-rs/redis-rs/pull/843, I'll implement the correct behavior once it is rebased.

Reply:
Ok, I think let's wait for #843 to go in first then.